text/1027-geo.rst

```diff
@@ -0,0 +1,295 @@
+  boundary and holes can be defined using ``LINESTRING``, ``CIRCULARSTRING``,
+  and ``COMPOUNDCURVE``, though.
+
+  .. code-block::
+
+    CURVEPOLYGON(
+      CIRCULARSTRING(0 0, 4 0, 4 4, 0 4, 0 0),
+      (1 1, 3 3, 3 1, 1 1) )
+
+All of these basic types are contiguous and describe some basic shapes. They
+can be combined into non-contiguous shapes by using a number of collection
+types:
+
+- Multi point representing a set of points.
+
+  E.g. ``MULTIPOINT ( (0 0), (1 2) )``
+
+- Multi line string representing a set of line strings.
+
+  E.g. ``MULTILINESTRING ( (0 0,1 1,1 2), (2 3,3 2,5 4) )``
+
+- Multi polygon representing a set of polygons.
+
+  E.g. ``MULTIPOLYGON (((1 5, 5 5, 5 1, 1 1, 1 5)), ((6 5, 9 1, 6 1, 6 5)))``
+
+- Multi curve representing a set of ``LINESTRING``, ``CIRCULARSTRING``, and
+  ``COMPOUNDCURVE``.
+
+  E.g. ``MULTICURVE( (0 0, 5 5), CIRCULARSTRING(4 0, 4 4, 8 4))``
+
+- Multi surface representing a set of ``POLYGON`` and ``CURVEPOLYGON`` values.
+
+Finally the most generic collection is simply *geometry collection*, which can
+contain any of the spacial types (including other geometry collections).
+
+E.g. ``GEOMETRYCOLLECTION ( POINT(2 3), LINESTRING(2 3, 3 4))``
+
+There are also a few more specialized types:
+
+- Polyhedral surface is a *contiguous* collection of polygons. So the polygons
+  must share edges.
+
+  .. code-block::
+
+    POLYHEDRALSURFACE Z (
+      ((0 0 0, 0 0 1, 0 1 1, 0 1 0, 0 0 0)),
+      ((0 0 0, 0 1 0, 1 1 0, 1 0 0, 0 0 0)),
+      ((0 0 0, 1 0 0, 1 0 1, 0 0 1, 0 0 0)),
+      ((1 1 0, 1 1 1, 1 0 1, 1 0 0, 1 1 0)),
+      ((0 1 0, 0 1 1, 1 1 1, 1 1 0, 0 1 0)),
+      ((0 0 1, 1 0 1, 1 1 1, 0 1 1, 0 0 1)) )
+
+- Triangle is a special closed polygon. It consists of 4 points. The points
+  must not be on the same line. First and last points must be identical.
+
+  E.g. ``TRIANGLE ((0 0, 0 9, 9 0, 0 0))``
+
+- TIN (Triangulated Irregular Network) is a collection of non-overlapping
+  triangles.
+
+  .. code-block::
+
+    TIN Z ( ((0 0 0, 0 0 1, 0 1 0, 0 0 0)), ((0 0 0, 0 1 0, 1 1 0, 0 0 0)) )
+
+From these types the ``geometry`` and ``geography`` values can be built up. In
+addition to the spacial data an SRID can be provided to determine the specific
+reference system being used (essentially giving concrete meaning to the
+coordinates). The default SRID for ``geometry`` is 0, which is plain Cartesian
+coordinates. The default SRID for ``geography`` is 4326 which uses longitude
+and latitude as coordinates.
+
+
+Proposed type hierarchy
+=======================
+
+We can start with the basic building block of the "point":
+
+.. code-block:: edgeql
+
+  CREATE ABSTRACT SCALAR TYPE gis::anygeo EXTENDING std::anyscalar;
+
+  CREATE SCALAR TYPE gis::point EXTENDING gis::anygeo;
+  CREATE SCALAR TYPE gis::pointz EXTENDING gis::anygeo;
+  CREATE SCALAR TYPE gis::pointzm EXTENDING gis::anygeo;
+
+  CREATE FUNCTION gis::point(
```

Comment:
Maybe we should reuse our `to_something()` naming convention here, so it's `postgis::to_point(...)`